notebooks/image_models/solutions/diffusion_model.ipynb

```diff
@@ -0,0 +1,723 @@
+{
```

Comment:
<p>Can you had some explanation before this cell about what this layer does and why we need it?</p><p> </p><p>This is a very long class. Is possible to break this class in smaller chunk and explain the methods? </p><p> </p><p>One way to do that would be to extract some of the methods (the most important) as standalone functions, comment on them in markdown cells, and reuse these functions to define the methods, making the class much shorter.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/347/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Reply:
<p>Broke apart sinembedding from other blocks and added markdown.</p>